pandas/tests/test_index.py

```diff
@@ -133,6 +133,14 @@ def test_str(self):
         self.assertTrue("'foo'" in str(idx))
         self.assertTrue(idx.__class__.__name__ in str(idx))
 
+    def test_repr_max_seq_item_setting(self):
```

Comment:
I am not fully following the logic of the test set up.

- on the one hand, you have the `self.indices` -> so iterating over these will run this test for all the different index types
- but `create_index` is also redefined in each subclass of `Base`, so in this way the test will also be run for the different index types

So it seems two ways to achieve a bit the same?